buildSrc/src/main/rocker/com/uber/okbuck/template/config/BuckDefs.rocker.raw

```diff
@@ -10,24 +10,83 @@ boolean useCompilationClasspath,
 String classpathExclusionRegex
 )
 
-import os, re
+def basename(path):
+    return path.rpartition("/")[-1]
 
+def dirname(path):
+    prefix, sep, _ = path.rpartition("/")
+    if not prefix:
+        return sep
+    else:
+        return prefix.rstrip("/")
+
+def paths_join(prefix, suffix):
+    result = prefix
+    if not prefix or prefix.endswith("/"):
+        result += suffix
+    else:
+        result += "/" + suffix
+    return result
 
 def merge_maps(**res_maps):
     result = {}
-    for path, res_map in res_maps.iteritems():
+    for path, res_map in res_maps.items():
         for key in res_map:
             if key in result and result[key] != res_map[key]:
-                dirname = os.path.dirname(key)
-                basename = os.path.basename(key)
-                new_key = '/'.join([dirname, path.replace('/', '_') + '_' + basename])
+                dir_name = dirname(key)
+                base_name = basename(key)
+                new_key = '/'.join([dir_name, path.replace('/', '_') + '_' + base_name])
                 result[new_key] = res_map[key]
             else:
                 result[key] = res_map[key]
 
     return result
 
 
+def subdir_glob(glob_specs, exclude = None, prefix = ""):
```

Comment:
can we first only replace the `subdir_glob` implementation and then move to full skylark?
this will allow us to verify that the rules name remains exactly the same

Reply:
also, probably move to it's own bazel file

Reply:
As tested, there is no behavioral change between this implementation and default one, unless SKYLARK is enabled as well. Source of behavioral change is map key ordering change.